nncf/experimental/torch/nas/bootstrapNAS/search/search.py

```diff
@@ -341,15 +343,19 @@ def get_macs_for_active_subnet() -> float:
 
         if self.best_config is not None:
             self._elasticity_ctrl.multi_elasticity_handler.activate_subnet_for_config(self.best_config)
-            self.bn_adaptation.run(self._model)
+            if self.bn_adaptation is not None:
+                self.bn_adaptation.run(self._model)
```

Comment:
This part doesn't seem working. Because Search Algorithm always instantiates `bn_adaptation`: https://github.com/openvinotoolkit/nncf/pull/1329/files#diff-7afaa162c96c4fe46d2bce9b80be7866d27d18e0914da3794e31cbabf007ff96R218-R220

Please add unit test checking that bn_adaptation is called or not called depending on num_samples provided in config. 
Here's the pseudocode:
```python
@pytest.mark.parametrize( ('bn_adapt_section', 'is_called'), 
(
    (section_with_zero_num_samples, False), 
    (section_with_non_zero_num_samples, True)
)
def test_bn_adapt(mocker, bn_adapt_section, is_called)
     nncf_config = NNCFConfig(...)
     update_bn_adapt_section(nncf_config, bn_adapt_section) # to be implemented
     mock_nncf_network = mocker.MagicMock(spec=NNCFNetwork)
     mock_ctrl =mocker..MagicMock(spec=ElasticityController)
     bn_adapt_run_patch = 
     mocker.patch('nncf.common.initialization.batchnorm_adaptation.BatchnormAdaptationAlgorithm.run')
     search_algo = SearchAlgorithm(mock_nncf_network, mock_ctrl, nncf_config)
     search_algo.run()
     if is_called:
          bn_adapt_run_patch.assert_called_once()
     else:
          bn_adapt_run_patch.assert_not_called()
```

Reply:
Thanks! Added test.